src/Calculator.tsx

```diff
@@ -0,0 +1,221 @@
+/* eslint-disable no-underscore-dangle */
+import React from 'react';
+
+enum Operator {
+  empty = '',
+  plus = 'plus',
+  minus = 'minus',
+  multiply = 'multiply',
+  divide = 'divide',
+}
+
+type State = {
+  prevNumber: null | number;
+  nextNumber: null | number;
+  operator: Operator;
+  result: string;
+};
+
+class Calculator extends React.Component<any, State> {
+  static localStorageKey = 'calculator-key';
+
+  static initialState = {
+    prevNumber: null,
+    nextNumber: null,
+    operator: Operator.empty,
+    result: '0',
+  };
+
+  static errorState = (errorMessage: string) => ({
+    ...Calculator.initialState,
+    result: errorMessage,
+  });
+
+  constructor(props: any) {
+    super(props);
+    const originalState = JSON.parse(localStorage.getItem(Calculator.localStorageKey) ?? '{}');
+    this.state = { ...Calculator.initialState, ...originalState };
+    window.addEventListener('beforeunload', this.saveCurrentStateBeforeLeave);
+    window.onbeforeunload = () => {
+      return 'Are you sure you want to leave?';
+    };
```

Comment:
```suggestion
    super(props);
    
    const originalState = JSON.parse(localStorage.getItem(Calculator.localStorageKey) ?? '{}');
    this.state = { ...Calculator.initialState, ...originalState };
    
    window.addEventListener('beforeunload', this.saveCurrentStateBeforeLeave);
    window.onbeforeunload = () => {
      return 'Are you sure you want to leave?';
    };
```
각 코드마다 같은 맥락인 코드들로만 묶어서 중간중간에 개행시켜주면 더 코드 읽기가 쉬워져요.